scripts/dependencies.py

```diff
@@ -0,0 +1,255 @@
+        constraint_match = re.match(version_constraint_regex, constraint)
+        checks.append(
+            partial_flip(
+                op_map[constraint_match.group("comparison_operator")],
+                constraint_match.group("version"),
+            )
+        )
+
+    return lambda v: all(c(v) for c in checks)
+
+
+def requirement_from_version_constraints(component, constraints):
+        meets_constraints = build_constraint_checker(constraints)
+        versions = (
+            version
+            for series in component["releases"]
+            for version in series["versions"]
+        )
+        for version in versions:
+            if meets_constraints(version["version"]):
+                requirement = {
+                    "name": component["name"],
+                    "ref": version["version"],
+                    "ref_type": "tag",
+                    "repo_url": component["repo_url"],
+                    "sha": version["sha"],
+                    "version": version["version"],
+                }
+                break
+        else:
+            raise Exception(
+                (
+                    "The component '{c_name}' has no version matching the "
+                    "constraints '{cs}'."
+                ).format(c_name=component["name"], cs=constraints)
+            )
+
+        return requirement
+
+
+def requirement_from_branch_constraints(component, constraints):
+    constraint = constraints.pop()
+    assert len(constraints) == 0
+
+    constraint_match = re.match(branch_constraint_regex, constraint)
+    branch_name = constraint_match.group("branch_name")
+    with TemporaryDirectory() as tmp_dir:
+        repo = git.Repo.clone_from(releases_repo, tmp_dir, branch=branch_name)
+        sha = repo.head.commit.hexsha
+
+    requirement = {
+        "name": component["name"],
+        "ref": branch_name,
+        "ref_type": "branch",
+        "repo_url": component["repo_url"],
+        "sha": sha,
+        "version": None
+    }
+
+    return requirement
+
+
+def update_requirements(metadata, component_dir):
+    requirements = {"dependencies": []}
+    for dependency in metadata["dependencies"]:
+        component = load_component(dependency["name"], component_dir)
+        constraints = dependency["constraints"]
+        try:
+            branch_constraints_schema.validate(constraints)
+        except SchemaError:
+            requirement = requirement_from_version_constraints(
+                component,
+                constraints
+            )
+        else:
+            requirement = requirement_from_branch_constraints(
+                component,
+                constraints
+            )
+
+        requirements["dependencies"].append(requirement)
+
+    return component_requirements_schema.validate(requirements)
+
+
+def download_requirements(requirements, dl_base_dir):
+    for requirement in requirements:
+        repo_dir = dl_base_dir + requirement["name"]
```

Comment:
`repo_dir = dl_base_base_dir + '/' + requirement["name"]`, incase someone doesn't add a trailing slash to `dl_base_dir`.

Reply:
Fixed with `os.path.join`.